sklearn/utils/_array_api.py

```diff
@@ -74,6 +93,13 @@ def _check_array_api_dispatch(array_api_dispatch):
                 "array_api_compat is required to dispatch arrays using the API"
```

Comment:
```suggestion
                "array-api-compat is required to dispatch arrays using the API"
```
Trying to be consistent with the spelling below. The package on PyPI is called "array-api-compat" so I think it makes sense to use that